File: AppData/Mods/BUILT-IN/_lua/LuaScript/Uzil/UI.lua
local this = {};

function this.inst (key)

	local inst = {};
	inst.key = key;

	UZAPI.UI.Inst(key);

	--[[ 銷毀 ]]
	inst.destroyInst = function ()
		this.destroyInst(inst.key);
	end

	--[[ 排序 ]]
	inst.sortInst = function (sort)
		this.sortInst(inst.key, sort);
	end

	--[[ 取得Canvas尺寸 ]]
	inst.getCanvasSize = function ()
		return this.getCanvasSize(inst.key);
	end

	--[[ 取得Canvas參考尺寸 ]]
	inst.getCanvasRefSize = function ()
		return this.getCanvasRefSize(inst.key);
	end

	--[[ 設置渲染模式 ]]
	inst.setRenderMode = function (renderMode)
		this.setRenderMode(inst.key, renderMode);
	end

	--[[ 設置 資料 ]]
	inst.create = function (id, data)
		this.create(inst.key, id, data);
	end

	--[[ 銷毀 ]]
	inst.remove = function (id)
		this.remove(inst.key, id);
	end

	--[[ 清空 ]]
	inst.clear = function ()
		this.clear(inst.key);
	end

	--[[ 強制更新 ]]
	inst.forceUpdate = function (id)
		this.forceUpdate(inst.key, id);
	end

	--[[ 排序 ]]
	inst.sort = function (id, sort)
		this.sort(inst.key, id, sort);
	end

	--[[ 設置 資料 ]]
	inst.setData = function (id, data)
		this.setData(inst.key, id, data);
	end

	--[[ 設置 位置 ]]
	inst.setPosition = function (id, pos_v2)
		this.setPosition(inst.key, id, pos_v2);
	end

	--[[ 取得 資料 ]]
	inst.getData = function (id, props)
		return this.getData(inst.key, id, props);
	end

	--[[ 設置 屬性 ]]
	inst.setProp = function (id, prop, value)
		this.setProp(inst.key, id, prop, value);
	end

	--[[ 設置 圖像資料 ]]
	inst.setImageData = function (id, data)
		this.setImageData(inst.key, id, data);
	end

	--[[ 設置 圖像屬性 ]]
	inst.setImageProp = function (id, prop, value)
		this.setImageProp(inst.key, id, prop, value);
	end

	--[[ 設置 文字資料 ]]
	inst.setTextData = function (id, data)
		this.setTextData(inst.key, id, data);
	end

	--[[ 設置 文字屬性 ]]
	inst.setTextProp = function (id, prop, value)
		this.setTextProp(inst.key, id, prop, value);
	end

	return inst;
end

--[[ 銷毀 ]]
function this.destroyInst (key)
	UZAPI.UI.DestroyInst(key);
end

--[[ 排序 ]]
function this.sortInst (key, sort)
	UZAPI.UI.SortInst(key, sort);
end

--[[ 取得Canvas尺寸 ]]
function this.getCanvasSize (key)
	return Json.decode(UZAPI.UI.GetCanvasSize(key));
end

--[[ 取得Canvas參考尺寸 ]]
function this.getCanvasRefSize (key)
	return Json.decode(UZAPI.UI.GetCanvasRefSize(key));
end

--[[ 設置渲染模式 ]]
function this.setRenderMode (key, renderMode)
	UZAPI.UI.SetRenderMode(key, renderMode);
end

--[[ 建立 ]]
function this.create (key, id, data)
	UZAPI.UI.Create(key, id, Json.encode(data));
end

--[[ 銷毀 ]]
function this.remove (key, id)
	UZAPI.UI.Remove(key, id);
end

--[[ 清空 ]]
function this.clear (key)
	UZAPI.UI.Clear(key);
end

--[[ 強制更新 ]]
function this.forceUpdate (key, id)
	UZAPI.UI.ForceUpdate(key, id);
end

--[[ 排序 ]]
function this.sort (key, id, sort)
	UZAPI.UI.Sort(key, id, sort);
end

--[[ 設置 資料 ]]
function this.setData (key, id, data)
	UZAPI.UI.SetData(key, id, Json.encode(data));
end

--[[ 取得 資料 ]]
function this.getData (key, id, props)
	return Json.decode(UZAPI.UI.GetData(key, id, Json.encode(props)));
end

--[[ 設置 位置 ]]
function this.setPosition (key, id, pos_v2)
	this.setData(key, id, {
		position = pos_v2
	});
end

--[[ 設置 屬性 ]]
function this.setProp (key, id, prop, value)
	if (type(value) == "table") then value = Json.encode(value) end
	UZAPI.UI.SetProp(key, id, prop, value);
end

--[[ 設置 圖像資料 ]]
function this.setImageData (key, id, data)
	UZAPI.UI.SetImageData(key, id, Json.encode(data));
end

--[[ 設置 圖像屬性 ]]
function this.setImageProp (key, id, prop, value)
	UZAPI.UI.SetImageProp(key, id, prop, value);
end

--[[ 設置 文字資料 ]]
function this.setTextData (key, id, data)
	UZAPI.UI.SetTextData(key, id, Json.encode(data));
end

--[[ 設置 文字屬性 ]]
function this.setTextProp (key, id, prop, value)
	UZAPI.UI.SetTextProp(key, id, prop, value);
end

--[[ 設置 動畫參數 ]]
function this.setAnimParam (key, id, paramName, value)
	UZAPI.UI.SetAnimParam(key, id, paramName, value);
end

return this;

File: Assets/Import_out/Uzil/API/UI.cs
using System.Collections.Generic;

using UnityEngine;

using Uzil;
using Uzil.UI;
using Uzil.BuiltinUtil;

using Uzil.ObjInfo;

namespace UZAPI {

public class UI {

	
	/*======================================Constructor==========================================*/

	/*=====================================Static Members========================================*/

	/*=====================================Static Funciton=======================================*/

	/** 建立 */
	public static void Inst (string instID) {
		UIMgr.Inst(instID);
	}

	/** 銷毀 */
	public static void DestroyInst (string instID) {
		UIMgr.DestroyInst(instID);
	}

	/** 排序 */
	public static void SortInst (string instID, float sort) {
		UIMgr.SortInst(instID, sort);
	}

	/** 設置 渲染模式 */
	public static void SetRenderMode (string instID, string renderMode) {
		if (renderMode == null) return;

		renderMode = renderMode.ToLower();
		
		RenderMode mode;
		
		switch (renderMode) {
			case "camera":
				mode = RenderMode.ScreenSpaceCamera;
				break;
			case "overlay":
				mode = RenderMode.ScreenSpaceOverlay;
				break;
			case "world":
				mode = RenderMode.WorldSpace;
				break;
			default:
				return;
		}
			
		UIMgr ui = UIMgr.Inst(instID);
		ui.SetRenderMode(mode);
	}

	/** 取得 Canvas 尺寸 */
	public static string GetCanvasSize (string instID) {
		CanvasObj canvas = CanvasUtil.Inst(instID);
		return DictSO.ToJson(DictSO.Vector2To(canvas.GetSize()));
	}

	/** 取得 Canvas 參考尺寸 */
	public static string GetCanvasRefSize (string instID) {
		CanvasObj canvas = CanvasUtil.Inst(instID);
		return DictSO.ToJson(DictSO.Vector2To(canvas.GetRefSize()));
	}

	/** 建立 */
	public static void Create (string instID, string id, string json) {
		UIMgr uiMgr = UIMgr.Inst(instID);
		DictSO data = DictSO.Json(json);
		uiMgr.Create(id, data);
	}

	/** 銷毀 */
	public static void Remove (string instID, string id) {
		UIMgr uiMgr = UIMgr.Inst(instID);
		uiMgr.Remove(id);
	}

	/** 清空 */
	public static void Clear (string instID) {
		UIMgr uiMgr = UIMgr.Inst(instID);
		uiMgr.Clear();
	}

	/** 強制更新 */
	public static void ForceUpdate (string instID, string id) {
		UIMgr uiMgr = UIMgr.Inst(instID);
		uiMgr.ForceUpdate(id);
	}

	/** 排序 */
	public static void Sort (string instID, string id, float sort) {
		UIMgr uiMgr = UIMgr.Inst(instID);
		uiMgr.Sort(id, sort);
	}

	/** 設置 資料 */
	public static void SetData (string instID, string id, string json) {
		try {
			UIMgr uiMgr = UIMgr.Inst(instID);
			DictSO data = DictSO.Json(json);
			uiMgr.SetData(id, data);
		} catch (System.Exception e) {
			Debug.LogError(e);
		}
	}

	/** 取得 資料 */
	public static string GetData (string instID, string id, string porps_json) {
		UIMgr uiMgr = UIMgr.Inst(instID);
		List<string> requests = DictSO.List<string>(porps_json);
		DictSO data = uiMgr.GetData(id, requests);
		return DictSO.ToJson(data);
	}

	/** 設置 屬性 */
	public static void SetProp (string instID, string id, string key, object value) {
		UIMgr uiMgr = UIMgr.Inst(instID);
		uiMgr.SetProp(id, key, value);
	}

	/** 設置 圖像資料 */
	public static void SetImageData (string instID, string id, string json) {
		UIMgr uiMgr = UIMgr.Inst(instID);
		ImageInfo imgInfo = new ImageInfo(json);
		DictSO data = (DictSO) imgInfo.ToMemo();
		uiMgr.SetImageData(id, data);
	}

	/** 設置 圖像屬性 */
	public static void SetImageProp (string instID, string id, string key, object value) {
		UIMgr uiMgr = UIMgr.Inst(instID);
		uiMgr.SetImageProp(id, key, value);
	}

	/** 設置 文字資料 */
	public static void SetTextData (string instID, string id, string json) {
		UIMgr uiMgr = UIMgr.Inst(instID);
		DictSO data = DictSO.Json(json);
		uiMgr.SetTextData(id, data);
	}

	/** 設置 文字屬性 */
	public static void SetTextProp (string instID, string id, string key, object value) {
		UIMgr uiMgr = UIMgr.Inst(instID);
		uiMgr.SetTextProp(id, key, value);
	}

	/** 設置 動畫參數 */
	public static void SetAnimParam (string instID, string id, string paramName, object value) {
		UIMgr uiMgr = UIMgr.Inst(instID);
		uiMgr.SetAnimParam(id, paramName, value);
	}

	/*=========================================Members===========================================*/

	/*========================================Components=========================================*/

	/*==========================================Event============================================*/

	/*======================================Unity Function=======================================*/
	
	/*========================================Interface==========================================*/

	/*=====================================Public Function=======================================*/

	/*===================================Protected Function======================================*/
	
	/*====================================Private Function=======================================*/
}


}
